[Dominic-Khoo] iP

Dominic-Khoo opened the pull request:
# Downy
> [!NOTE]
> "Simplicity is the soul of efficiency."  - Austin Freeman (in _The Eye of Osiris_) ([source](https://vanderburg.org/etc/soft-quotes.html))

Downy is a chatbot that epitomises the concept of efficiency through simplicity. It is,
- Text-based
- Quick
- ~~Beginner~~ EXTREMELY beginner friendly

All you have to do is:

1. Fork and pull from [here](https://github.com/Dominic-Khoo/ip)
2. Run `Downy.java` using your IDE
3. Enter your tasks
4. Let it manage your tasks for you :grinning:

And it is **FREE**!

Features: 

- [x] Managing various tasks
- [ ] Reminders (coming soon)
- [ ] Calendar (coming soon)

Here's the `main` method for reference:
```
public static void main(String[] args) {
        new Downy("data/tasks.txt").run();
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Dominic-Khoo committed:
Update name to Downy across all files

Dominic-Khoo committed:
Add echoing of user input and exit command

Dominic-Khoo committed:
Add list to store user inputs

Dominic-Khoo committed:
Add task management

Dominic-Khoo committed:
Add different event types

Dominic-Khoo committed:
Add testing using I/O redirection technique

Dominic-Khoo committed:
Add exception handling

Dominic-Khoo committed:
Change TaskList to ArrayList<Task>

Dominic-Khoo committed:
Add delete function

Dominic-Khoo committed:
Add save function

Dominic-Khoo committed:
Merge branch 'branch-Level-7'

Dominic-Khoo committed:
Change string representation of date and time with LocalDateTime

Dominic-Khoo committed:
Merge branch 'branch-Level-8'

Dominic-Khoo committed:
Modify code to practice better object oriented programming

Dominic-Khoo committed:
Merge branch 'branch-MoreOOP'

Dominic-Khoo committed:
Package files

Dominic-Khoo committed:
Merge branch 'branch-packages'

Dominic-Khoo committed:
Merge branch 'add-gradle-support'

Dominic-Khoo committed:
Add Gradle support for Downy

Dominic-Khoo committed:
Add test cases for Storage, Parser, Ui, TaskList

Dominic-Khoo committed:
Merge branch 'branch-junit'

Dominic-Khoo committed:
Add jar package functionality

Dominic-Khoo committed:
Add JavaDoc documentation

Dominic-Khoo committed:
Edit to adhere to Java SE-EDU coding standard

Dominic-Khoo committed:
Add find command

Dominic-Khoo committed:
Merge branch 'branch-A-JavaDoc'

Dominic-Khoo committed:
Merge branch 'branch-A-CodingStandard'

Dominic-Khoo committed:
Merge branch 'branch-Level-9'

Dominic-Khoo committed:
Bug fix for Parser and Event

BlazeChron commented on src/test/java/parser/ParserTest.java:
```diff
@@ -0,0 +1,149 @@
+import exceptions.InvalidFormatException;
+import exceptions.MissingArgumentException;
+
+public class ParserTest {
```
Perhaps this could be ParserTester instead?

BlazeChron commented on src/main/java/ui/Ui.java:
```diff
@@ -0,0 +1,170 @@
+    /**
+     * Displays a welcome message to the user.
+     */
+    public void showWelcome() {
```
Could this be homogenised to showWelcomeMessage()?

BlazeChron commented on src/main/java/storage/Storage.java:
```diff
@@ -0,0 +1,270 @@
+     *
+     * @param t The Todo task to be written to the storage file.
+     */
+    public void writeTodoToFile(Todo t) {
```
I like the descriptive names used here

BlazeChron commented on src/main/java/tasks/Deadline.java:
```diff
@@ -0,0 +1,46 @@
+ */
+public class Deadline extends Task {
+
+    private final LocalDateTime time;
```
Perhaps a more specialised name for the time variable could be more appropriate?

BlazeChron reviewed (commented):
The code is great. Only a few nits.

Dominic-Khoo committed:
Update checkStyle for main and test folders

Dominic-Khoo committed:
Merge branch 'A-CheckStyle'

Raghava-Chittidi commented on src/main/java/exceptions/InvalidCommandException.java:
```diff
@@ -0,0 +1,17 @@
+ * when an invalid command is entered by the user. This exception extends
+ * {@code DownyException}.
+ */
+public class InvalidCommandException extends DownyException {
```
I like how you created different exceptions for different errors

Raghava-Chittidi commented on src/main/java/storage/Storage.java:
```diff
@@ -0,0 +1,272 @@
+            // Read all lines from the file
+            List<String> lines = Files.readAllLines(Paths.get(this.filePath));
+
+            // Filter out the line corresponding to the task to be deleted
```
I like how you add comments at each section of the code to explain what it is doing

Raghava-Chittidi commented on src/main/java/storage/Storage.java:
```diff
@@ -0,0 +1,272 @@
+                    }
+                }
+                default -> {
+                    // Do nothing
```
Maybe an error can be thrown here? The file might be possibly corrupted if some other letter is present, triggering the default case

Raghava-Chittidi commented on text-ui-test/EXPECTED.TXT:
```diff
@@ -1,7 +1,70 @@
+________________________________________
+exceptions.InvalidCommandException: Invalid command entered
+Here are a list of valid commands:
+ - list
```
I like how you show the list of valid commands when a user has entered an invalid command

Raghava-Chittidi reviewed (commented):
Overall, the code LGTM! It is really well done and includes a lot of tests and well-written Javadoc comments. I also agree with the things @BlazeChron has already said

Dominic-Khoo committed:
Add GUI to Downy

Dominic-Khoo committed:
Edit checkstyle for Main and MainWindow

Dominic-Khoo committed:
Merge branch 'branch-Level-10'

Dominic-Khoo committed:
Add assertions

Dominic-Khoo committed:
Add InvalidFileFormatException to enhance code quality

To improve code quality, utilised default branch in
case statements to detect errors.

Additionally, ensure that no empty catch blocks are
present.

Dominic-Khoo committed:
Edit parse method in Parser class

Previously, parse method included switch-case for each
command, with additional low-level details.

Extract method refactoring was done to adhere to
SLAP, as a means to enhance code quality.

Dominic-Khoo committed:
Merge pull request #1 from Dominic-Khoo/branch-A-Assertions

Add assertions

Dominic-Khoo committed:
Merge branch 'master' of https://github.com/Dominic-Khoo/ip

Dominic-Khoo committed:
Merge pull request #2 from Dominic-Khoo/branch-A-CodeQuality

Improve code quality of Downy

Dominic-Khoo committed:
Merge branch 'master' of https://github.com/Dominic-Khoo/ip

Dominic-Khoo committed:
Add notes function to Downy

Dominic-Khoo committed:
Merge branch 'branch-D-Notes'

Dominic-Khoo committed:
Add more error handling for Downy

Dominic-Khoo committed:
Merge branch 'branch-A-MoreErrorHandling'

Dominic-Khoo committed:
Add more testing to Downy based on updated features

Dominic-Khoo committed:
Merge branch 'branch-A-MoreTesting'

Dominic-Khoo committed:
Add user guide to Downy
